Code/Tools/AssetProcessor/native/InternalBuilders/SettingsRegistryBuilder.cpp

```diff
@@ -430,7 +431,10 @@ namespace AssetProcessor
                     }
                     file.Close();
 
-                    response.m_outputProducts.emplace_back(outputPath, m_assetType, productSubID + aznumeric_cast<AZ::u32>(i));
+                    AZ::u32 hashedSpecialization = static_cast<AZ::u32>(AZStd::hash<AZStd::string_view>{}(specializationString));
+                    AZ_Assert(hashedSpecialization != 0, "Product ID generation failed for specialization %s. This can result in a product ID collision with other builders for this asset.",
+                        specializationString.data());
```

Comment:
You need to use `%.*s` for the string format and the `AZ_STRING_ARG(specializationString)` parameter for output a string_view safely into a variadic string format.

This is because a string_view might not be null-terminated and we need it to tell use how many characters to print

Reply:
gotcha, I've fixed it